editor/store/effects.js

```diff
@@ -254,6 +253,8 @@ export default {
 		dispatch( removeNotice( TRASH_POST_NOTICE_ID ) );
 		apiRequest( { path: `/wp/v2/${ basePath }/${ postId }`, method: 'DELETE' } ).then(
 			() => {
+				const post = getCurrentPost( getState() );
+				dispatch( resetPost( { ...post, status: 'trashed' } ) );
```

Comment:
The correct status is `'trash'`.

https://codex.wordpress.org/Function_Reference/get_post_status#Return_Values